homework-assignments-1/task-4.ts

```diff
@@ -0,0 +1,169 @@
+// KEPT EVERYTHING IN A SINGLE FILE FOR EASIER READING
+
+// 1. Product Class
+class Product {
+  constructor(
+    public name: string,

```

Comment:
You shouldn't expose class members as public. due to security reasons it must be private and you must access them only by getters and setters